### PR TITLE
Replace speed generation without waiting by Time to first token (TTFT)

### PR DESCRIPTION
**Pull request by maxDavid40**
- Replace all object python about speed generation without waiting by TTFT's objects
- Add a third axe on speed_generation_plot.png because TTFT does not have same scale than other curves
- Update the README.md

### CODE REVIEW

**Review by mfournioux — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ After the bench suite ends, you obtain a folder containing :
...
+   - A folder `speed_generation` containing, for each couple of input length/output length, a graph showing the speed generation (per request) in token per second vs the number of parallel requests. The curve is the usual speed generation ie the number of tokens divided by the time taken to generate them. On the graph is also shown the time to get the first token generated in millisecondes and the max kv cache usage for this number of parallel requests. The corresponding data is in the `data` folder
```
Comment (gsolard) [inline README.md:48]: - A folder `speed_generation` containing, for each couple of input length/output length, a graph showing the speed generation (per request) in token per second vs the number of parallel requests. The graph also shows the time to the first token generated in milliseconds and the max kv cache usage for this number of parallel requests. The corresponding data is in the `data` folder

File: src/benchmark_llm_serving/make_graphs.py [hunk 2]
```diff
@@ -141,47 +141,51 @@ def make_speed_generation_graph_for_one_input_output(input_length: int, output_l
...
-        special_speed_lower_percentiles.append(data_summary[parallel_requests_nb][name_special_speed][0])
-        special_speed_upper_percentiles.append(data_summary[parallel_requests_nb][name_special_speed][2])
+        speed_time_to_first_token_plot.append(data_summary[parallel_requests_nb][name_TTFT][1])
+        speed_time_to_first_token_lower_percentiles.append(data_summary[parallel_requests_nb][name_TTFT][0])
+        speed_time_to_first_token_upper_percentiles.append(data_summary[parallel_requests_nb][name_TTFT][2])
         max_kv_cache.append(data_summary[parallel_requests_nb]['max_kv_cache'])
 
     # Figure definition
     fig, ax1 = plt.subplots()
+    fig.subplots_adjust(right=0.75)
```
Comment (gsolard) [inline src/benchmark_llm_serving/make_graphs.py:160]: If this provides a better graph, shouldn't we put it for all the graphs ?

File: src/benchmark_llm_serving/make_graphs.py [hunk 3]
```diff
@@ -113,7 +113,7 @@ def make_speed_generation_graph_for_one_input_output(input_length: int, output_l
     lower_percentile = 'percentile_25'
     upper_percentile = 'percentile_75'
     name_speed = "speed_generation"
-    name_special_speed = "speed_without_waiting_time"
+    name_TTFT = 'speed_time_to_first_token'
```
[the remark this answers is not in the document]
Comment (maxDavid40) [reply src/benchmark_llm_serving/make_graphs.py]: Fixe : [b24675caeeb43ed747013c5cdcc98f7f96d0da39](https://github.com/France-Travail/benchmark_llm_serving/pull/11/commits/b24675caeeb43ed747013c5cdcc98f7f96d0da39)

File: README.md (hunk 1 above)
Comment (maxDavid40) [reply README.md]: Fix : [c9b1465b46d96114acbf2ebb1ebf3951fbdc7038](https://github.com/France-Travail/benchmark_llm_serving/pull/11/commits/c9b1465b46d96114acbf2ebb1ebf3951fbdc7038)
Indeed, the second sentence is useless

**Review by gsolard — APPROVED** [review]

**Pull request merged**